src/msac.rs

```diff
@@ -282,8 +280,16 @@ pub unsafe extern "C" fn dav1d_msac_decode_symbol_adapt_c(
     }
     return val;
 }
-#[no_mangle]
-pub unsafe extern "C" fn dav1d_msac_decode_bool_adapt_c(
+
+unsafe extern "C" fn dav1d_msac_decode_symbol_adapt_c(
+    s: *mut MsacContext,
+    cdf: *mut uint16_t,
+    n_symbols: size_t,
+) -> libc::c_uint {
+    dav1d_msac_decode_symbol_adapt_rust(s, cdf, n_symbols)
+}
```

Comment:
Why do we still need this `_c` version of the function? I would assume this method is only used internally right? So we should just be able to call the `_rust` version directly right?

Reply:
The `_c` version is needed for use in an `unsafe extern "C"` `fn` ptr that roundtrips through asm.

Reply:
Oh that's fun. I guess we now know for sure that the asm does indeed sometimes call back into C/Rust.

Reply:
Interesting indeed. Might be a good idea to leave a comment lest we forget and muck with the signature.

Reply:
We'll get compile errors from the `fn` ptr `symbol_adapt16` field if we try to change it.

I'm not actually certain the asm calls `.symbol_adapt16`, but it uses other things passed in through `MsacContext`, so I'm guessing it does.